Yoast/wordpress-seo
# Add tests for the memoizers

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Adds tests for the meta tags context memoizer and the presentation memoizer.

## Relevant technical choices:

* I created doubles for both memoizers in order to access the cache.
* We used to use `makePartial()` to make a class's own functions. Now, we are no longer using that approach, but still we don't want to test all of a class functions in one big test. Therefore, for the meta tags context I created a mock function (`mock_get`) to mock one function that was called by another function in the same class. Of course, the `get` function also has its own dedicated tests.
* This PR introduces a few CS errors on class name length, which I couldn't prevent. Therefore, I fixed CS errors in unrelated files to meet the threshold again.

## Test instructions

This PR can be tested by following these steps:

* See Travis pass.
* See that code coverage for the two memoizers is at 100%.

## UI changes

## Documentation

## Quality assurance


Fixes https://yoast.atlassian.net/browse/P2-237
